compiler/src/dotty/tools/dotc/transform/MoveStatics.scala

```diff
@@ -60,6 +61,7 @@ class MoveStatics extends MiniPhase with SymTransformer {
       def move(module: TypeDef, companion: TypeDef): List[Tree] = {
         assert(companion ne module)
         if (!module.symbol.is(Flags.Module)) move(companion, module)
+        else if (companion.ne(null) && companion.symbol.is(Flags.Trait)) List(module, companion)
```

Comment:
You can just write `!= null`, it translates to the same code as `ne(null)`. I realize the orginal version uses `ne` as well.